### ISSUE TITLE
Can read-only images on NFS be supported?

### ISSUE THREAD

**Issue by reidpr** [hpc/charliecloud#9]
Investigate.

See #7 and #8.

**Comment by reidpr** [comment]
See also relevant [FAQ entry](https://hpc.github.io/charliecloud/faq.html#ch-run-fails-with-can-t-re-mount-image-read-only).

**Comment by reidpr** [comment]
I wonder if `MS_BIND` is the problem here.

**Comment by j-ogas** [comment]
@olifre had suggested using `MS_NOSUID` to resolve #401. It worked. Interestingly enough, I've noticed that is also works for NFS!

Without `MS_NOSUID` flag (current behavior):
```
$ stat -f -L -c %T ../centos7/
nfs
$ ./bin/ch-run ../centos7/ -- /bin/true
ch-run[51805]: can't re-mount image read-only (is it on NFS?): Operation not permitted (charliecloud.c:230 1)
```

With `MS_NOSUID` flag:

```
$ stat -f -L -c %T ../centos7/
nfs
$ ./bin/ch-run ../centos7/ -- /bin/true
$ echo $?
0
```

As mentioned in 401, the addition of this flag doesn't appear to produce regressions in the test suite.

**Comment by reidpr** [comment]
Excellent detective work!

I believe my misunderstanding when I wrote that code some time ago was, I expected the other mount options to remain unchanged when we re-mounted R/O.

I think the right answer is probably to copy over the existing mount options, since according to the docs cited in the Moby bug, there are four other mount options that could cause the same failure, and it could be that further mount options should remain unchanged to avoid surprises.

**Comment by reidpr** [comment]
OK, here's a proposed plan:

1. Use `statvfs(2)` to get the existing mount options, which we modify and pass back to `mount(2)`.
2. Remove the now-obsolete FAQ entry on NFS (currently §6.2.3).
3. The PR closes both this issue and #401 (but I don't think they're duplicates).
